queue_job/job.py

```diff
@@ -561,7 +561,11 @@ def store(self):
 
         db_record = self.db_record()
         if db_record:
-            db_record.with_context(_job_edit_sentinel=edit_sentinel).write(vals)
+            uid = self.env.context.get('uid')
+            if not uid:
+                db_record.with_context(_job_edit_sentinel=edit_sentinel).write(vals)
+            else:
+                db_record.with_context(_job_edit_sentinel=edit_sentinel).with_user(uid).write(vals)
```

Comment:
In order to avoid duplicating a long line could you use a simplified mode with extra variable, please?

I mean,

```suggestion
            db_record_wctx = db_record.with_context(_job_edit_sentinel=edit_sentinel)
            if uid:
                db_record_wctx = db_record_wctx.with_user(uid)
            db_record_wctx.write(vals)
```

Reply:
Done